src/main/java/seedu/duke/ui/TextUi.java

```diff
@@ -0,0 +1,128 @@
+package seedu.duke.ui;
+
+import java.nio.BufferOverflowException;
+import java.util.NoSuchElementException;
+import java.util.Scanner;
+
+/**
+ * TextUI class for reading user inputs and printing outputs.<br/>
+ * Subclasses of TextUI class can override separator, printErrorFor and printOutputMessage.<br/>
+ * This is to accommodate to the uniqueness of each feature.
+ */
+public class TextUi {
+    private static final String ALERT_SEPARATOR = "!!!!!!----------!!!!!!!----------!!!!!!!";
+    private static final String INDENTATION_SPACES = "    ";
+    private static final int DEFAULT_SEPARATOR_LENGTH = 40;
+    private static final Scanner SCANNER = new Scanner(System.in);
+
+    private String separator = "-";
+
+    /**
+     * Print spaces before output message for better formatting.
+     */
+    public void printIndentation() {
+        System.out.print(INDENTATION_SPACES);
+    }
+
+    /**
+     * Read user's input command and return back the command string.<br/>
+     *
+     * @return user input command with leading/dangling whitespace being removed
+     */
+    public String getCommand() {
+        String userCommand = "";
+        try {
+            String inputLine = SCANNER.nextLine();
+            userCommand = inputLine.trim();
+        } catch (BufferOverflowException bufferOverFlowException) {
+            printErrorFor(bufferOverFlowException);
+        } catch (NoSuchElementException noElementException) {
+            printErrorFor(noElementException);
+        }
+        return userCommand;
+    }
+
+    /**
+     * Customise separators for each feature.<br/>
```

Comment:
Add newline after `<br>` there must be explicit newlines between paragraphs within the JavaDoc also